task/sonarqube-scanner/0.2/README.md

```diff
@@ -0,0 +1,180 @@
+# SonarQube
+
+SonarQube™ is the leading tool for continuously inspecting the Code Quality and Security™ of your codebases, all while empowering development teams. Analyze over 25 popular programming languages including C#, VB.Net, JavaScript, TypeScript and C++. It detects bugs, vulnerabilities and code smells across project branches and pull requests.
+
+The following task can be used to perform static analysis on the source code provided the SonarQube server is hosted.
+
+For creating your own `sonar-project.properties` please follow the guide [here](https://docs.sonarqube.org/latest/analysis/analysis-parameters/). Sample properties file can be found [here](./examples/sonar-project.properties)
+
+## Install the Task
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/sonarqube-scanner/0.2/sonarqube-scanner.yaml
+```
+
+## Pre-requisite
+
+Install the `git-clone` task from the catalog
+
+```
+https://raw.githubusercontent.com/tektoncd/catalog/v1beta1/git/git-clone.yaml
+```
+
+## Parameters
+
+- **SONAR_HOST_URL**: Host URL where the sonarqube server is running
+- **SONAR_PROJECT_KEY**: Project's unique key
+
+> _Note_ : Parameters are provided in that case when we want to override the corresponding values in `sonar-project.properties` or there is no `sonar-project.properties` present for the project which needs to be analyzed
+
+## Workspaces
+
+- **source-dir**: `PersistentVolumeClaim`-type so that volume can be shared among git-clone and sonarqube task. Sample PVC can be found [here](../0.2/samples/pvc.yaml)
+- **sonar-settings**: To mount the `sonar-project.properties` via the `ConfigMap`. (_Default_ : `emptyDir:{}`)
```

Comment:
in the task this workspace is marked as optional so can we document this over here as well?